### ISSUE TITLE
mwoffliner drops downloaded images from zim creation if imagemin toolchain fails processing them

### ISSUE THREAD

**Issue by cm8** [openzim/mwoffliner#1980]
https://github.com/openzim/mwoffliner/blob/ad5dc1d6071552c1a9b577fa1a619cfba0fe6938/src/Downloader.ts#L504-L521

The catch in line 505 seems to miss an error handler:  If the imagemin call fails, then a successfully downloaded image will be dropped from ZIM creation.  mwoffliner can do better in these cases: if the size reduction fails, log and take the downloaded original.

-----
As an example for a possibly failing  image see
https://upload.wikimedia.org/wikipedia/commons/thumb/1/1e/Schleiflade_Tontraktur_Animation.gif/120px-Schleiflade_Tontraktur_Animation.gif

Prior to kohler/gifsicle@0e02f7b62a9a3a344c34f92ddb7e178ad3b3e3ff  this file would fail a gifsicle run and thus not be included in any zim even if the resource successfully downloaded..

**Comment by kelson42** [comment]
@cm8 Good catch if confirmed, we will have a look soon this your issue... or do you have a patch in mind ?

**Comment by kelson42** [comment]
@benoit74 I'm not 100% sure current behaviour is wrong. Should we just:
* Keep it like it is
* Print a warning in the log

What do you think?

**Comment by benoit74** [comment]
Keeping current behavior means that we will drop any animated GIF for instance.
But adding original image means that the ZIM might get quite big.

I'm puzzled.

**Comment by benoit74** [comment]
For the time being, I recommend to fail scrape if image conversion fails. There is no known reasons for this to happen, and it will cause scraper to upload bad original image to S3 and we won't know anymore we have uncompressed images in the ZIMs.

**Comment by benoit74** [comment]
(and scraper already achieves to optimize animated GIFs, [Orgue](https://browse.library.kiwix.org/viewer#wikipedia_fr_all_maxi_2026-05/Orgue) article works fine with the animated GIF
